### DISCUSSION TITLE
example of chatgpt for synchro of 2 tracks

### DISCUSSION THREAD

**Discussion by tinolayco** [katspaugh/wavesurfer.js#3516] [Q&A]
is it possible to wavesurfer to sync 2 tracks together as chatgpt exemple ?

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Audio Player</title>
    <style>
        .audio-container {
            margin-bottom: 20px;
        }

        .audio-seek-bar,
        .volume-control {
            width: 100%;
        }

        .controls-container {
            display: flex;
            justify-content: space-between;
            align-items: center;
        }
    </style>
</head>
<body>
    <div class="audio-container">
        <audio class="audio-player" src="./sounds/Canary In A Coalmine.mp4"></audio>
    </div>

    <div class="audio-container">
        <audio class="audio-player" src="./sounds/vocals.mp3"></audio>
    </div>

    <div class="controls-container">
        <div>
            <input type="range" id="volumeControl1" min="0" max="1" step="0.1" value="1" class="volume-control">
        </div>
        <div>
            <input type="range" id="volumeControl2" min="0" max="1" step="0.1" value="1" class="volume-control">
        </div>
    </div>

    <input type="range" id="commonSeekBar" min="0" value="0" step="1" class="audio-seek-bar">
    <button class="play-button">Play</button>

    <script>
        document.addEventListener('DOMContentLoaded', function () {
            var audioPlayers = document.querySelectorAll('.audio-player');
            var commonSeekBar = document.getElementById('commonSeekBar');
            var volumeControls = document.querySelectorAll('.volume-control');
            var playButton = document.querySelector('.play-button');

            audioPlayers.forEach(function (audioPlayer, index) {
                audioPlayer.addEventListener('timeupdate', function() {
                    updateSeekBar();
                });

                audioPlayer.addEventListener('ended', function() {
                    // Pause et remise à zéro de la barre de défilement à la fin de chaque piste
                    audioPlayers.forEach(function(player) {
                        player.pause();
                        player.currentTime = 0;
                    });
                    commonSeekBar.value = 0;
                });
            });

            commonSeekBar.addEventListener('input', function() {
                seekCommonAudio();
            });

            playButton.addEventListener('click', function() {
                audioPlayers.forEach(function(player) {
                    player.play();
                });
            });

            volumeControls.forEach(function(volumeControl, index) {
                volumeControl.addEventListener('input', function() {
                    setVolume(index, volumeControl.value);
                });
            });

            function updateSeekBar() {
                // Utiliser la position de seek de la première piste comme référence
                var currentTime = audioPlayers[0].currentTime;
                var duration = audioPlayers[0].duration;

                // Mettre à jour la barre de défilement
                commonSeekBar.value = currentTime;
            }

            function seekCommonAudio() {
                var seekTime = commonSeekBar.value;

                // Appliquer la position de seek à chaque piste
                audioPlayers.forEach(function(player) {
                    player.currentTime = seekTime;
                });
            }

            function setVolume(index, volume) {
                audioPlayers[index].volume = volume;
            }
        });
    </script>
</body>
</html>

**Comment by katspaugh** [comment]
You can play two tracks simultaneously using wavesurfer-multitrack: https://wavesurfer.xyz/examples/?a33e2fa96802a35e1bdb125a6c0c5517
